cpp/src/command_classes/AVSimple.cpp

```diff
@@ -0,0 +1,162 @@
+#include "command_classes/CommandClasses.h"
+#include "command_classes/MultiInstance.h"
+#include "command_classes/AVSimple.h"
+#include "command_classes/AVSimpleCommandItem.h"
+#include "Defs.h"
+#include "Msg.h"
+#include "Node.h"
+#include "Driver.h"
+#include "platform/Log.h"
+#include <vector>
+
+#include "value_classes/ValueShort.h"
+#include "value_classes/ValueList.h"
+
+using namespace OpenZWave;
+
+uint8 m_sequence;
+
+enum SimpleAVCmd
+{
+	SimpleAVCmd_Set = 0x01
+};
+
+//-----------------------------------------------------------------------------
+// <AVSimple::HandleMsg>
+// Handle a message from the Z-Wave network
+//-----------------------------------------------------------------------------
+bool AVSimple::HandleMsg
+(
+	uint8 const* _data,
+	uint32 const _length,
+	uint32 const _instance
+)
+{
+	return true;
+}
+
+//-----------------------------------------------------------------------------
+// <AVSimple::SetValue>
+// Set a value in the Z-Wave device
+//-----------------------------------------------------------------------------
+bool AVSimple::SetValue
+(
+	Value const& _value
+)
+{
+	uint16 shortval;
+	if (ValueID::ValueType_Short == _value.GetID().GetType())
+	{
+		ValueShort const* value = static_cast<ValueShort const*>(&_value);
+		shortval = value->GetValue();
+	}
+	else if (ValueID::ValueType_List == _value.GetID().GetType())
+	{
+		ValueList const* value = static_cast<ValueList const*>(&_value);
+		shortval = value->GetItem()->m_value;
+	}
+	else return false;
+
+	uint8 instance = _value.GetID().GetInstance();
+	Msg* msg = new Msg("SimpleAVCmd_Set", GetNodeId(), REQUEST, FUNC_ID_ZW_SEND_DATA, true);
+	msg->SetInstance(this, instance);
+	msg->Append(GetNodeId());
+	msg->Append(8); // Length
+	msg->Append(GetCommandClassId());
+	msg->Append(SimpleAVCmd_Set);
+	msg->Append(m_sequence++); // Crutch
+	msg->Append(0);
+	msg->Append(0);
+	msg->Append(0);
+	msg->Append(shortval >> 8);
+	msg->Append(shortval & 0xff);
+	msg->Append(GetDriver()->GetTransmitOptions());
+	GetDriver()->SendMsg(msg, Driver::MsgQueue_Send);
+
+	if (m_sequence == 255)
+		m_sequence = 0;
+
+	return true;
+}
+
+//-----------------------------------------------------------------------------
+// <AVSimple::CreateVars>
+// Create the values managed by this command class
+//-----------------------------------------------------------------------------
+void AVSimple::CreateVars
+(
+	uint8 const _instance
+)
+{
+	if (Node* node = GetNodeUnsafe())
+	{
+		// Create list value
+		vector<ValueList::Item> items;
```

Comment:
Shouldn't we implement SIMPLE_AV_CONTROL_GET and SIMPLE_AV_CONTROL_SUPPORTED_* commands to figure out what keys are actually supported by the device?